File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> cl = new HashMap<>();
        
        for (int i = 0; i < clothes.length; i++) {
            String type = clothes[i][1];
            cl.put(type, cl.getOrDefault(type, 0) + 1);
        }
        
        int answer = 1;
        for (int value : cl.values()) {
            answer *= (value + 1);
        }
        answer -= 1;
        
        return answer;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    cl = {}
    for i in range(len(clothes)):
        cl[clothes[i][1]] = cl.get(clothes[i][1], 0) + 1
    
    answer = 1
    print(cl)
    for value in cl.values():
        answer *= (value + 1)
    print(answer, len(cl))
    answer -= 1
    
    return answer